### PR TITLE
Quarantine Improvements

### PR DESCRIPTION
**Pull request by Ashod**
- wsd: quarantine: refactor file listing
- wsd: quarantine: initialize config values only once
- wsd: quarantine: early check for valid file
- wsd: quarantine: better config defaults
- wsd: quarantine: use the file before renaming
- wsd: quarantine: Entry and filename parsing
- wsd: quarantine: use Entry to track files
- wsd: early sleep-for-debugger
- wsd: quarantine: initialization
- wsd: quarantine: calculate quarantine size
- wsd: quarantine: deleteOldQuarantineVersions
- wsd: quarantine: makeQuarantineSpace
- wsd: quarantine: remove unused functions

### CODE REVIEW

File: coolwsd.xml.in [hunk 1]
```diff
@@ -287,10 +287,10 @@
     </monitors>
 
     <quarantine_files desc="Files are stored here to be examined later in cases of crashes or similar situation." default="false" enable="false">
-        <limit_dir_size_mb desc="Maximum directory size. On exceeding the specified limit, older files will be deleted." default="250" type="uint"></limit_dir_size_mb>
-        <max_versions_to_maintain desc="How many versions of the same file to keep." default="2" type="uint"></max_versions_to_maintain>
+        <limit_dir_size_mb desc="Maximum directory size, in MBs. On exceeding the specified limit, older files will be deleted." default="250" type="uint"></limit_dir_size_mb>
+        <max_versions_to_maintain desc="How many versions of the same file to keep." default="5" type="uint"></max_versions_to_maintain>
         <path desc="Absolute path of the directory under which quarantined files will be stored. Do not use a relative path." type="path" relative="false"></path>
-        <expiry_min desc="Time in mins after quarantined files will be deleted." type="int" default="30"></expiry_min>
+        <expiry_min desc="Time in mins after quarantined files will be deleted." type="int" default="3000"></expiry_min>
```
Comment (timar) [inline coolwsd.xml.in:293]: Why did you change it here, and not at https://github.com/CollaboraOnline/online/blob/07f73b429e457ef550fc257011876c13e4a1b193/wsd/QuarantineUtil.cpp#L51 ?

Comment (Ashod) [reply coolwsd.xml.in:294]: Thanks for the catch; I had missed them. Fixed.

**Review by timar — APPROVED** [review]
